src/main.ts

```diff
@@ -56,8 +59,8 @@ const proxyEmitter = new eventEmitter()
 // Used mainly to avoid starting a new proxy when closing the active one on shutdown
 let appShuttingDown: boolean = false
 let currentProxyProcess: ProxyProcess | null
-let proxyReady = false
-export let proxyPort = 6000
+let proxyStatus: ProxyStatus = 'offline'
+export let appSettings: AppSettings
```

Comment:
- Proxy status will be tracked across its life cycle so we can _observe_ it in the UI.
- All the app configuration now lives in `appSettings`. This variable is dynamic and will change as settings are changed.